### PR TITLE
Version 2 updates

### PR DESCRIPTION
**Pull request by TieDyeGeek**
* Now works with smaller BaseEntity objects (from Core version 2.0) so your tables no longer have to have all the properties from BaseEntity
* Removed CachedRepository
* Expanded ReadOnlyRepository to accept Query or Expression based filtering for almost all functions
* Queries that returned IEnumerable<> now return List<>
* Create, Update, and Delete operations can now accept List<TEntity>
* Delete can now accept an Expression to delete one or many entities
* QueryBuilder now has a WithId extension for IEntityWithId based objects

### CODE REVIEW

File: src/CSESoftware.Repository/IRepository.cs [hunk 1]
```diff
@@ -1,22 +1,37 @@
...
         void Update<TEntity>(TEntity entity)
-            where TEntity : class, IBaseEntity;
+            where TEntity : class, IEntity;
+
+        void Update<TEntity>(List<TEntity> entities)
+            where TEntity : class, IEntity;
 
-        void Delete<TEntity>(object id)
-            where TEntity : class, IBaseEntity;
+        void Delete<TEntity, T>(T id)
```
Comment (NickBehrends) [inline src/CSESoftware.Repository/IRepository.cs:23]: What does the implementation/usage for this look like?

Comment (TieDyeGeek) [reply src/CSESoftware.Repository/IRepository.cs:23]: _repository.Delete<Pizza, int>(1);

Comment (NickBehrends) [reply src/CSESoftware.Repository/IRepository.cs:23]: Is that better than _repository.Delete<Pizza>(1)? What value do we get from specifying the type? I have not looked at implementation yet so maybe it's used there.

Comment (TieDyeGeek) [reply src/CSESoftware.Repository/IRepository.cs:23]: This is do to Core now taking the type of an ID rather than just making it object

**Review by jared-craig-cse — APPROVED** [review]

**Review by NickBehrends — APPROVED** [review]

**Pull request merged**